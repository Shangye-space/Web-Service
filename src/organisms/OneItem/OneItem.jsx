import React, { useState, useEffect } from 'react';
import axios from 'axios';
import classnames from 'classnames';
import ItemCard from '../../atoms/ItemCard';
import Loader from '../../atoms/Loader';

import styles from './OneItem.module.scss';

const OneItem = ({ id }) => {
    const [isLoading, setLoading] = useState(true);
    let [item, setItem] = useState([]);
    let [info, setInfo] = useState([]);

    useEffect(() => {
        axios
            .get(`http://localhost:3348/api/item/${id}`)
            .then(res => {
                setItem((item = res.data[0]));
                setLoading(false);
            })
            .catch(err => console.log(err.message));

        axios.get(`http://localhost:3348/api/item_info/${id}`).then(res => {
            setInfo((info = res.data[0]));
        });
    }, []);

    return (
        <div>
            {isLoading ? (
                <div
                    style={{
                        height: 1000,
                        display: 'flex',
                        justifyContent: 'center',
                        alignContent: 'center',
                    }}
                >
                    <Loader />
                </div>
            ) : (
                <div className={classnames('container', styles.wrapper)}>
                    <ItemCard
                        name={item.Name}
                        price={item.Price}
                        id={item.ID}
                    />
                    <div className="col-3">
                        <span>{info.Quantity}</span>
                        <br />
                        <span>{info.Description}</span>
                        <br />
                        <span>{info.Size}</span>
                        <br />
                        <span>{info.Color}</span>
                        <br />
                        <span>{info.Manufacturer}</span>
                        <br />
                        <span>{info.ItemCode}</span>
                        <br />
                        <span>{info.Material}</span>
                    </div>
                </div>
            )}
        </div>
    );
};

export default OneItem;
