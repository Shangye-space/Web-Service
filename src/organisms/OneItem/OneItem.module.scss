.wrapper {
    display: flex;
    justify-content: center;
    flex-wrap: wrap;

    padding: 1em 0;
    padding-top: 5.5em;
}
