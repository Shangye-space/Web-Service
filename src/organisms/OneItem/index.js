import OneItem from './OneItem.jsx';

export default OneItem;
