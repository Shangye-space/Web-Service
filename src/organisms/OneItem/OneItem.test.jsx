import React from 'react';
import ReactDOM from 'react-dom';
import OneItem from './OneItem.jsx';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<OneItem />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('snapshot renders', () => {
    const component = renderer.create(<OneItem />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
