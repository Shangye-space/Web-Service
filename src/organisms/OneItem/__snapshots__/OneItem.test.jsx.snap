// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders 1`] = `
<div>
  <div
    style={
      Object {
        "alignContent": "center",
        "display": "flex",
        "height": 1000,
        "justifyContent": "center",
      }
    }
  >
    <span
      className="loader"
    />
  </div>
</div>
`;
